sound/soc/sof/ipc4-pcm.c

```diff
@@ -600,6 +600,15 @@ static void sof_ipc4_build_time_info(struct snd_sof_dev *sdev, struct snd_sof_pc
 		}
 	}
 
+	if (!host_copier || !dai_copier) {
+		dev_err(sdev->dev, "host or dai copier are not found");
+
+		kfree(spcm->private);
+		spcm->private = NULL;
```

Comment:
do we actually need to free here?

Reply:
If it is not free it here, time_info is used for delay but it is not initialized.

````c
static snd_pcm_sframes_t sof_ipc4_pcm_delay(struct snd_soc_component *component,
					    struct snd_pcm_substream *substream)
{
        ...
	stream = &spcm->stream[substream->stream];
	time_info = stream->private;
	if (!time_info)
		return 0;

       // use time_info
````